WavesCS/Transactions/DataTransaction.cs

```diff
@@ -12,13 +13,34 @@ public class DataTransaction : Transaction
         public decimal Fee { get; }
 
         private const byte Version = 1;
-        
+
         public DataTransaction(byte[] senderPublicKey, Dictionary<string, object> entries,
             decimal? fee = null) : base(senderPublicKey)
         {
             Entries = entries;
             Fee = fee ?? ((GetBody().Length + 70) / 1024 + 1) * 0.001m;
         }
+      
+        public DataTransaction(Dictionary<string, object> tx) : base(tx)
+        {
+            Entries = tx.Get<DictionaryObject[]>("data")
+                        .ToDictionary(entry => entry["key"].ToString(), entry =>
+                        {
+                            var value = entry["value"].ToString();
```

Comment:
We don't need to concert it to string, better to take appropriate type depending in value type.